src/EprPrnIntegration.Api/Functions/EmailNpwdReconciliationFunction.cs

```diff
@@ -7,77 +7,83 @@
 using Microsoft.Extensions.Logging;
 using Microsoft.Extensions.Options;
 using System.Text;
+using EprPrnIntegration.Common.RESTServices.BackendAccountService.Interfaces;
+using EprPrnIntegration.Common.RESTServices.PrnBackendService.Interfaces;
 
 namespace EprPrnIntegration.Api.Functions;
 
 public class EmailNpwdReconciliationFunction(
-    IEmailService _emailService,
-    IAppInsightsService _appInsightsService,
-    IOptions<FeatureManagementConfiguration> _featureConfig,
-    ILogger<EmailNpwdReconciliationFunction> _logger
-)
+    IEmailService emailService,
+    IAppInsightsService appInsightsService,
+    IOptions<FeatureManagementConfiguration> featureConfig,
+    ILogger<EmailNpwdReconciliationFunction> logger,
+    IUtilities utilities,
+    IPrnService prnService)
 {
     [Function("EmailNpwdReconciliation")]
     public async Task Run([TimerTrigger("%EmailNpwdReconciliationTrigger%")] TimerInfo myTimer)
     {
-        var isOn = _featureConfig.Value.RunIntegration ?? false;
+        var isOn = featureConfig.Value.RunIntegration ?? false;
         if (!isOn)
         {
-            _logger.LogInformation("EmailNpwdReconciliation function is disabled by feature flag");
+            logger.LogInformation("EmailNpwdReconciliation function is disabled by feature flag");
             return;
         }
 
-        await EmailNpwdIssuedPrnsReconciliationAsync();
+        var issuedPrnEmailTask = EmailNpwdIssuedPrnsReconciliationAsync();
+        var updatedPrnEmailTask = EmailUpdatedPrnReconciliationAsync();
 
+        // Execute tasks concurrently, ensuring one task can continue even if the other fails
+        await Task.WhenAll(issuedPrnEmailTask, updatedPrnEmailTask);
     }
 
-    public async Task EmailNpwdIssuedPrnsReconciliationAsync()
+    public async Task EmailUpdatedPrnReconciliationAsync()
     {
-        _logger.LogInformation("EmailNpwdIssuedPrnsReconciliationAsync function executed at: {ExecutionDateTime}", DateTime.UtcNow);
-
-        string csv = string.Empty;
-        List<ReconcileIssuedPrn> prns = new List<ReconcileIssuedPrn>();
+        logger.LogInformation("EmailUpdatedPrnReconciliationAsync function executed at: {ExecutionDateTime}", DateTime.UtcNow);
 
         try
         {
-            // retrieve custom events from Application Insights
-            prns = await _appInsightsService.GetIssuedPrnCustomEventLogsLast24hrsAsync();
-
-            // generate csv
-            csv = TransformPrnsToCsv(prns);
-
-            // send reconciliation email with link to csv file
-            _emailService.SendIssuedPrnsReconciliationEmailToNpwd(DateTime.UtcNow, prns.Count, csv);
+            var updatedPrns = await prnService.GetReconsolidatedUpdatedPrns();
+            var csvData = new Dictionary<string, List<string>>
```

Comment:
I may be missing something but don't you have to loop on updatedPrns and add them into custom events selecting from it?